packages/core/src/geometry/GeometrySystem.js

```diff
@@ -227,6 +234,32 @@ export default class GeometrySystem extends System
         }
     }
 
+    /**
+     * Takes a geometry and program and generates a unique signature for them.
+     *
+     * @param {PIXI.Geometry} geometry to get signature from
+     * @param {PIXI.Program} prgram to test geometry against
+     * @returns {String} Unique signature of the geometry and program
+     * @private
+     */
+    getSignature(geometry, program)
+    {
+        const attribs = geometry.attributes;
+        const shaderAttributes = program.attributeData;
+
+        const strings = [geometry.id];
```

Comment:
Make that array global. Dont spawn new ones on cached signatures. set length to 0 before return.